packages/block-library/src/social-links/edit.js

```diff
@@ -1,43 +1,124 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
 /**
  * WordPress dependencies
  */
 
 import { Fragment } from '@wordpress/element';
 
 import {
+	BlockControls,
 	__experimentalUseInnerBlocksProps as useInnerBlocksProps,
 	useBlockProps,
 	InspectorControls,
 } from '@wordpress/block-editor';
-import { ToggleControl, PanelBody } from '@wordpress/components';
+import {
+	Dropdown,
+	MenuGroup,
+	MenuItem,
+	PanelBody,
+	ToggleControl,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { check } from '@wordpress/icons';
+import { DOWN } from '@wordpress/keycodes';
 
 const ALLOWED_BLOCKS = [ 'core/social-link' ];
 
+const sizeOptions = [
+	{ name: __( 'Small' ), value: 'has-small-icon-size' },
+	{ name: __( 'Normal' ), value: 'has-normal-icon-size' },
+	{ name: __( 'Large' ), value: 'has-large-icon-size' },
+	{ name: __( 'Huge' ), value: 'has-huge-icon-size' },
+];
+
 export function SocialLinksEdit( props ) {
 	const {
-		attributes: { openInNewTab },
+		attributes: { iconSize, openInNewTab },
 		setAttributes,
 	} = props;
 
 	const SocialPlaceholder = (
 		<div className="wp-block-social-links__social-placeholder">
-			<div className="wp-block-social-link wp-social-link-facebook"></div>
-			<div className="wp-block-social-link wp-social-link-twitter"></div>
-			<div className="wp-block-social-link wp-social-link-instagram"></div>
```

Comment:
why is the `block-` removed?

Reply:
The frontend markup has the following CSS classes applied:

`<li class="wp-social-link wp-social-link-wordpress wp-block-social-links has-large-icon-size is-style-pill-shape">`

In effort to simply get the frontend and editor closer, it made sense to me to keep the classes more closely to the frontend.

Reply:
Yes, I'm confused about when the block prefix is added and when excluded

Reply:
I cleaned up a reference to `wp-block-social-link` so all CSS now refers to only `wp-social-link`. I'm not quite sure why we have both those classes, I'll defer to @mkaz.

Reply:
FWIW, the `wp-block-social-links` is only in children (li elements) because of a bug we just fixed https://github.com/WordPress/gutenberg/pull/26344

Reply:
Marks this as resolved, per Andres comment.